src/Core/Grid/Query/Builder/CustomerQueryBuilder.php

```diff
@@ -0,0 +1,201 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query\Builder;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Class CustomerQueryBuilder builds queries to fetch data for customers grid.
+ */
+final class CustomerQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var int
+     */
+    private $contextShopId;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @var int[]
+     */
+    private $contextShopIds;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param int $contextShopId
+     * @param int $contextLangId
+     * @param int[] $contextShopIds
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        $contextShopId,
+        $contextLangId,
+        $contextShopIds
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->contextShopId = $contextShopId;
+        $this->contextLangId = $contextLangId;
+        $this->contextShopIds = $contextShopIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $searchQueryBuilder = $this->getCustomerQueryBuilder($searchCriteria)
+            ->select('c.id_customer, c.firstname, c.lastname, c.email, c.active, c.newsletter, c.optin')
+            ->addSelect('c.date_add, gl.name as social_title, s.name as shop_name, c.company')
+        ;
+
+        $this->appendTotalSpentQuery($searchQueryBuilder);
+        $this->appendLastVisitQuery($searchQueryBuilder);
+
+        return $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $countQueryBuilder = $this->getCustomerQueryBuilder($searchCriteria)
+            ->select('COUNT(*)')
+        ;
+
+        return $countQueryBuilder;
+    }
+
+    private function getCustomerQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'customer', 'c')
+            ->leftJoin(
+                'c',
+                $this->dbPrefix . 'gender_lang',
+                'gl',
+                'c.id_gender = gl.id_gender AND gl.id_lang = :context_lang_id'
+            )
+            ->leftJoin(
+                'c',
+                $this->dbPrefix . 'shop',
+                's',
+                'c.id_shop = s.id_shop'
+            )
+            ->where('c.deleted = 0')
```

Comment:
`c.deleted is FALSE` => `'deleted' => array('type' => self::TYPE_BOOL, 'validate' => 'isBool', 'copy_post' => false),`